src/storage/src/hummock/compactor/context.rs

```diff
@@ -75,7 +78,60 @@ impl CompactorContext {
             memory_limiter: MemoryLimiter::unlimit(),
             task_progress_manager: Default::default(),
             await_tree_reg: None,
-            running_task_count: Arc::new(AtomicU32::new(0)),
+            running_task_parallelism: Arc::new(AtomicU32::new(0)),
+            max_task_parallelism: Arc::new(AtomicU32::new(u32::MAX)),
+        }
+    }
+
+    pub fn acquire_task_quota(&self, parallelism: u32) -> bool {
+        let mut running_u32 = self.running_task_parallelism.load(Ordering::SeqCst);
+        let max_u32 = self.max_task_parallelism.load(Ordering::SeqCst);
+
+        if parallelism + running_u32 > max_u32 {
+            return false;
+        }
```

Comment:
This branch is no longer needed because it is covered by the while loop below.

Reply:
fix, PTAL